compiler/plugins/api_web.ml

```diff
@@ -26,20 +25,36 @@ open Lcalc.Backends
 open Lcalc.To_ocaml
 module D = Dcalc.Ast
 
-let name = "jsoo"
+let name = "api_web"
 let extension = ".ml"
 
+(** Contains all format functions used to generating the [js_of_ocaml] wrapper
+    of the corresponding Catala program. *)
 module To_jsoo = struct
+  let to_camel_case (s : string) : string =
+    String.split_on_char '_' s
+    |> (function
+         | hd :: tl -> hd :: List.map String.capitalize_ascii tl | l -> l)
+    |> String.concat ""
+
+  let format_struct_field_name_camel_case
+      (fmt : Format.formatter)
+      (v : Dcalc.Ast.StructFieldName.t) : unit =
+    let s =
+      Format.asprintf "%a" Dcalc.Ast.StructFieldName.format_t v
+      |> to_ascii |> to_lowercase |> avoid_keywords |> to_camel_case
+    in
+    Format.fprintf fmt "%s" s
+
   let format_tlit (fmt : Format.formatter) (l : Dcalc.Ast.typ_lit) : unit =
     Dcalc.Print.format_base_type fmt
       (match l with
       | TUnit -> "unit"
       | TInt -> "int"
-      | TRat -> "Js.number Js.t"
-      | TMoney -> "Js.number Js.t"
+      | TRat | TMoney -> "Js.number Js.t"
       | TDuration -> "Runtime_jsoo.Runtime.duration Js.t"
       | TBool -> "bool Js.t"
-      | TDate -> "Js.date Js.t")
+      | TDate -> "Js.js_string Js.t")
```

Comment:
Why are you doing that ? Isn't it better to keep it as a `Js.date` ?

Reply:
Because that's [how it's done](https://json-schema.org/understanding-json-schema/reference/string.html?highlight=date#dates-and-times) with JSON schemas.